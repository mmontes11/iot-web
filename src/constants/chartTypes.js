export const BARCHART = "BARCHART";
export const LINECHART = "LINECHART";
