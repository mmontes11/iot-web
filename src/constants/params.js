export const FIRST_PARAM = "FIRST_PARAM";
export const SECOND_PARAM = "SECOND_PARAM";
