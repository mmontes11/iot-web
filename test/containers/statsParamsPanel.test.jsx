import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { MemoryRouter } from "react-router-dom";
import StatsParamsPanel from "containers/statsParamsPanel";
import { initialState } from "../constants";

describe("containers/statsParamsPanel", () => {
  it("renders in initial state", () => {
    const store = configureStore([thunk])(initialState);
    const wrapper = shallow(
      <MemoryRouter initialEntries={["/"]} keyLength={0}>
        <Provider store={store}>
          <StatsParamsPanel onParamsSelected={() => undefined} onReset={() => undefined} />
        </Provider>
      </MemoryRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders before selecting type", () => {
    const state = {
      ...initialState,
      params: {
        ...initialState.params,
        type: {
          items: ["event", "measurement"],
          isActive: true,
          selectedItem: null,
        },
      },
    };
    const store = configureStore([thunk])(state);
    const wrapper = shallow(
      <MemoryRouter initialEntries={["/"]} keyLength={0}>
        <Provider store={store}>
          <StatsParamsPanel onParamsSelected={() => undefined} onReset={() => undefined} />
        </Provider>
      </MemoryRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders when type is selected", () => {
    const state = {
      ...initialState,
      params: {
        ...initialState.params,
        type: {
          items: ["event", "measurement"],
          isActive: false,
          selectedItem: "event",
        },
      },
    };
    const store = configureStore([thunk])(state);
    const wrapper = shallow(
      <MemoryRouter initialEntries={["/"]} keyLength={0}>
        <Provider store={store}>
          <StatsParamsPanel onParamsSelected={() => undefined} onReset={() => undefined} />
        </Provider>
      </MemoryRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders when loading observations", () => {
    const state = {
      ...initialState,
      params: {
        ...initialState.params,
        observation: {
          items: [],
          isActive: false,
          isLoading: true,
          isDisabled: true,
          selectedItem: null,
        },
      },
    };
    const store = configureStore([thunk])(state);
    const wrapper = shallow(
      <MemoryRouter initialEntries={["/"]} keyLength={0}>
        <Provider store={store}>
          <StatsParamsPanel onParamsSelected={() => undefined} onReset={() => undefined} />
        </Provider>
      </MemoryRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders before selecting observation", () => {
    const state = {
      ...initialState,
      params: {
        ...initialState.params,
        observation: {
          items: ["temperature", "humidity"],
          isActive: false,
          isLoading: false,
          isDisabled: false,
          selectedItem: null,
        },
      },
    };
    const store = configureStore([thunk])(state);
    const wrapper = shallow(
      <MemoryRouter initialEntries={["/"]} keyLength={0}>
        <Provider store={store}>
          <StatsParamsPanel onParamsSelected={() => undefined} onReset={() => undefined} />
        </Provider>
      </MemoryRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders before selecting observation", () => {
    const state = {
      ...initialState,
      params: {
        ...initialState.params,
        observation: {
          items: ["temperature", "humidity"],
          isActive: false,
          isLoading: false,
          isDisabled: false,
          selectedItem: null,
        },
      },
    };
    const store = configureStore([thunk])(state);
    const wrapper = shallow(
      <MemoryRouter initialEntries={["/"]} keyLength={0}>
        <Provider store={store}>
          <StatsParamsPanel onParamsSelected={() => undefined} onReset={() => undefined} />
        </Provider>
      </MemoryRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders when observation is selected", () => {
    const state = {
      ...initialState,
      params: {
        ...initialState.params,
        observation: {
          items: ["temperature", "humidity"],
          isActive: false,
          isLoading: false,
          isDisabled: false,
          selectedItem: "temperature",
        },
      },
    };
    const store = configureStore([thunk])(state);
    const wrapper = shallow(
      <MemoryRouter initialEntries={["/"]} keyLength={0}>
        <Provider store={store}>
          <StatsParamsPanel onParamsSelected={() => undefined} onReset={() => undefined} />
        </Provider>
      </MemoryRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
