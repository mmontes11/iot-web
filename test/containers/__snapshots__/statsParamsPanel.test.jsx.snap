// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/statsParamsPanel renders in initial state 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/",
    ]
  }
  keyLength={0}
>
  <Connect(Connect(ResetOnUnmount))
    onParamsSelected={[Function]}
    onReset={[Function]}
  />
</MemoryRouter>
`;

exports[`containers/statsParamsPanel renders when TYPE is selected 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/",
    ]
  }
  keyLength={0}
>
  <Connect(Connect(ResetOnUnmount))
    onParamsSelected={[Function]}
    onReset={[Function]}
  />
</MemoryRouter>
`;

exports[`containers/statsParamsPanel renders when loading TYPE 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/",
    ]
  }
  keyLength={0}
>
  <Connect(Connect(ResetOnUnmount))
    onParamsSelected={[Function]}
    onReset={[Function]}
  />
</MemoryRouter>
`;
